#' Genomics API Objects 
#' Upload, process, query, and search Genomics data in the cloud.
#' 
#' Auto-generated code by googleAuthR::gar_create_api_objects
#'  at 2017-03-05 19:53:39
#' filename: /Users/mark/dev/R/autoGoogleAPI/googlegenomicsv1.auto/R/genomics_objects.R
#' api_json: api_json
#' 
#' Objects for use by the functions created by googleAuthR::gar_create_api_skeleton

#' ExportVariantSetRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The variant data export request.
#' 
#' @param callSetIds If provided, only variant call information from the specified call sets
#' @param projectId Required
#' @param format The format for the exported data
#' @param bigqueryDataset Required
#' @param bigqueryTable Required
#' 
#' @return ExportVariantSetRequest object
#' 
#' @family ExportVariantSetRequest functions
#' @export
ExportVariantSetRequest <- function(callSetIds = NULL, projectId = NULL, format = NULL, 
    bigqueryDataset = NULL, bigqueryTable = NULL) {
    structure(list(callSetIds = callSetIds, projectId = projectId, format = format, 
        bigqueryDataset = bigqueryDataset, bigqueryTable = bigqueryTable), class = "gar_ExportVariantSetRequest")
}

#' SearchAnnotationsRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param pageToken The continuation token, which is used to page through large result sets
#' @param pageSize The maximum number of results to return in a single page
#' @param start The start position of the range on the reference, 0-based inclusive
#' @param annotationSetIds Required
#' @param referenceName The name of the reference to query, within the reference set associated
#' @param referenceId The ID of the reference to query
#' @param end The end position of the range on the reference, 0-based exclusive
#' 
#' @return SearchAnnotationsRequest object
#' 
#' @family SearchAnnotationsRequest functions
#' @export
SearchAnnotationsRequest <- function(pageToken = NULL, pageSize = NULL, start = NULL, 
    annotationSetIds = NULL, referenceName = NULL, referenceId = NULL, end = NULL) {
    structure(list(pageToken = pageToken, pageSize = pageSize, start = start, annotationSetIds = annotationSetIds, 
        referenceName = referenceName, referenceId = referenceId, end = end), class = "gar_SearchAnnotationsRequest")
}

#' OperationEvent Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' An event that occurred during an Operation.
#' 
#' @param endTime Optional time of when event finished
#' @param startTime Optional time of when event started
#' @param description Required description of event
#' 
#' @return OperationEvent object
#' 
#' @family OperationEvent functions
#' @export
OperationEvent <- function(endTime = NULL, startTime = NULL, description = NULL) {
    structure(list(endTime = endTime, startTime = startTime, description = description), 
        class = "gar_OperationEvent")
}

#' CodingSequence Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param start The start of the coding sequence on this annotation's reference sequence,
#' @param end The end of the coding sequence on this annotation's reference sequence,
#' 
#' @return CodingSequence object
#' 
#' @family CodingSequence functions
#' @export
CodingSequence <- function(start = NULL, end = NULL) {
    structure(list(start = start, end = end), class = "gar_CodingSequence")
}

#' TestIamPermissionsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' Response message for `TestIamPermissions` method.
#' 
#' @param permissions A subset of `TestPermissionsRequest
#' 
#' @return TestIamPermissionsResponse object
#' 
#' @family TestIamPermissionsResponse functions
#' @export
TestIamPermissionsResponse <- function(permissions = NULL) {
    structure(list(permissions = permissions), class = "gar_TestIamPermissionsResponse")
}

#' GetIamPolicyRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' Request message for `GetIamPolicy` method.
#' 
#' 
#' 
#' @return GetIamPolicyRequest object
#' 
#' @family GetIamPolicyRequest functions
#' @export
GetIamPolicyRequest <- function() {
    list()
}

#' SearchReferencesResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param references The matching references
#' @param nextPageToken The continuation token, which is used to page through large result sets
#' 
#' @return SearchReferencesResponse object
#' 
#' @family SearchReferencesResponse functions
#' @export
SearchReferencesResponse <- function(references = NULL, nextPageToken = NULL) {
    structure(list(references = references, nextPageToken = nextPageToken), class = "gar_SearchReferencesResponse")
}

#' SearchAnnotationSetsRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param pageToken The continuation token, which is used to page through large result sets
#' @param pageSize The maximum number of results to return in a single page
#' @param datasetIds Required
#' @param types If specified, only annotation sets that have any of these types are
#' @param name Only return annotations sets for which a substring of the name matches this
#' @param referenceSetId If specified, only annotation sets associated with the given reference set
#' 
#' @return SearchAnnotationSetsRequest object
#' 
#' @family SearchAnnotationSetsRequest functions
#' @export
SearchAnnotationSetsRequest <- function(pageToken = NULL, pageSize = NULL, datasetIds = NULL, 
    types = NULL, name = NULL, referenceSetId = NULL) {
    structure(list(pageToken = pageToken, pageSize = pageSize, datasetIds = datasetIds, 
        types = types, name = name, referenceSetId = referenceSetId), class = "gar_SearchAnnotationSetsRequest")
}

#' SearchReadGroupSetsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The read group set search response.
#' 
#' @param nextPageToken The continuation token, which is used to page through large result sets
#' @param readGroupSets The list of matching read group sets
#' 
#' @return SearchReadGroupSetsResponse object
#' 
#' @family SearchReadGroupSetsResponse functions
#' @export
SearchReadGroupSetsResponse <- function(nextPageToken = NULL, readGroupSets = NULL) {
    structure(list(nextPageToken = nextPageToken, readGroupSets = readGroupSets), 
        class = "gar_SearchReadGroupSetsResponse")
}

#' LinearAlignment Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A linear alignment can be represented by one CIGAR string. Describes themapped position and local alignment of the read to the reference.
#' 
#' @param mappingQuality The mapping quality of this alignment
#' @param position The position of this alignment
#' @param cigar Represents the local alignment of this sequence (alignment matches, indels,
#' 
#' @return LinearAlignment object
#' 
#' @family LinearAlignment functions
#' @export
LinearAlignment <- function(mappingQuality = NULL, position = NULL, cigar = NULL) {
    structure(list(mappingQuality = mappingQuality, position = position, cigar = cigar), 
        class = "gar_LinearAlignment")
}

#' SearchReferencesRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param md5checksums If present, return references for which the
#' @param accessions If present, return references for which a prefix of any of
#' @param pageToken The continuation token, which is used to page through large result sets
#' @param referenceSetId If present, return only references which belong to this reference set
#' @param pageSize The maximum number of results to return in a single page
#' 
#' @return SearchReferencesRequest object
#' 
#' @family SearchReferencesRequest functions
#' @export
SearchReferencesRequest <- function(md5checksums = NULL, accessions = NULL, pageToken = NULL, 
    referenceSetId = NULL, pageSize = NULL) {
    structure(list(md5checksums = md5checksums, accessions = accessions, pageToken = pageToken, 
        referenceSetId = referenceSetId, pageSize = pageSize), class = "gar_SearchReferencesRequest")
}

#' Dataset Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A Dataset is a collection of genomic data.For more genomics resource definitions, see [Fundamentals of GoogleGenomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics)
#' 
#' @param projectId The Google Cloud project ID that this dataset belongs to
#' @param id The server-generated dataset ID, unique across all datasets
#' @param createTime The time this dataset was created, in seconds from the epoch
#' @param name The dataset name
#' 
#' @return Dataset object
#' 
#' @family Dataset functions
#' @export
Dataset <- function(projectId = NULL, id = NULL, createTime = NULL, name = NULL) {
    structure(list(projectId = projectId, id = id, createTime = createTime, name = name), 
        class = "gar_Dataset")
}

#' ImportVariantsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The variant data import response.
#' 
#' @param callSetIds IDs of the call sets created during the import
#' 
#' @return ImportVariantsResponse object
#' 
#' @family ImportVariantsResponse functions
#' @export
ImportVariantsResponse <- function(callSetIds = NULL) {
    structure(list(callSetIds = callSetIds), class = "gar_ImportVariantsResponse")
}

#' ReadGroup Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A read group is all the data that's processed the same way by the sequencer.
#' 
#' @param ReadGroup.info The \link{ReadGroup.info} object or list of objects
#' @param sampleId A client-supplied sample identifier for the reads in this read group
#' @param datasetId The dataset to which this read group belongs
#' @param experiment The experiment used to generate this read group
#' @param name The read group name
#' @param referenceSetId The reference set the reads in this read group are aligned to
#' @param info A map of additional read group information
#' @param id The server-generated read group ID, unique for all read groups
#' @param predictedInsertSize The predicted insert size of this read group
#' @param programs The programs used to generate this read group
#' @param description A free-form text description of this read group
#' 
#' @return ReadGroup object
#' 
#' @family ReadGroup functions
#' @export
ReadGroup <- function(ReadGroup.info = NULL, sampleId = NULL, datasetId = NULL, experiment = NULL, 
    name = NULL, referenceSetId = NULL, info = NULL, id = NULL, predictedInsertSize = NULL, 
    programs = NULL, description = NULL) {
    structure(list(ReadGroup.info = ReadGroup.info, sampleId = sampleId, datasetId = datasetId, 
        experiment = experiment, name = name, referenceSetId = referenceSetId, info = info, 
        id = id, predictedInsertSize = predictedInsertSize, programs = programs, 
        description = description), class = "gar_ReadGroup")
}

#' ReadGroup.info Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A map of additional read group information. This must be of the formmap<string, string[]> (string key mapping to a list of string values).
#' 
#' 
#' 
#' @return ReadGroup.info object
#' 
#' @family ReadGroup functions
#' @export
ReadGroup.info <- function() {
    list()
}

#' ReadGroupSet Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A read group set is a logical collection of read groups, which arecollections of reads produced by a sequencer. A read group set typicallymodels reads corresponding to one sample, sequenced one way, and aligned oneway.* A read group set belongs to one dataset.* A read group belongs to one read group set.* A read belongs to one read group.For more genomics resource definitions, see [Fundamentals of GoogleGenomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics)
#' 
#' @param ReadGroupSet.info The \link{ReadGroupSet.info} object or list of objects
#' @param info A map of additional read group set information
#' @param id The server-generated read group set ID, unique for all read group sets
#' @param datasetId The dataset to which this read group set belongs
#' @param readGroups The read groups in this set
#' @param filename The filename of the original source file for this read group set, if any
#' @param name The read group set name
#' @param referenceSetId The reference set to which the reads in this read group set are aligned
#' 
#' @return ReadGroupSet object
#' 
#' @family ReadGroupSet functions
#' @export
ReadGroupSet <- function(ReadGroupSet.info = NULL, info = NULL, id = NULL, datasetId = NULL, 
    readGroups = NULL, filename = NULL, name = NULL, referenceSetId = NULL) {
    structure(list(ReadGroupSet.info = ReadGroupSet.info, info = info, id = id, datasetId = datasetId, 
        readGroups = readGroups, filename = filename, name = name, referenceSetId = referenceSetId), 
        class = "gar_ReadGroupSet")
}

#' ReadGroupSet.info Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A map of additional read group set information.
#' 
#' 
#' 
#' @return ReadGroupSet.info object
#' 
#' @family ReadGroupSet functions
#' @export
ReadGroupSet.info <- function() {
    list()
}

#' SearchVariantSetsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The search variant sets response.
#' 
#' @param nextPageToken The continuation token, which is used to page through large result sets
#' @param variantSets The variant sets belonging to the requested dataset
#' 
#' @return SearchVariantSetsResponse object
#' 
#' @family SearchVariantSetsResponse functions
#' @export
SearchVariantSetsResponse <- function(nextPageToken = NULL, variantSets = NULL) {
    structure(list(nextPageToken = nextPageToken, variantSets = variantSets), class = "gar_SearchVariantSetsResponse")
}

#' Empty Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A generic empty message that you can re-use to avoid defining duplicatedempty messages in your APIs. A typical example is to use it as the requestor the response type of an API method. For instance:    service Foo {      rpc Bar(google.protobuf.Empty) returns (google.protobuf.Empty);    }The JSON representation for `Empty` is empty JSON object `{}`.
#' 
#' 
#' 
#' @return Empty object
#' 
#' @family Empty functions
#' @export
Empty <- function() {
    list()
}

#' Entry Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param status The creation status
#' @param annotation The created annotation, if creation was successful
#' 
#' @return Entry object
#' 
#' @family Entry functions
#' @export
Entry <- function(status = NULL, annotation = NULL) {
    structure(list(status = status, annotation = annotation), class = "gar_Entry")
}

#' Position Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' An abstraction for referring to a genomic position, in relation to somealready known reference. For now, represents a genomic position as areference name, a base number on that reference (0-based), and adetermination of forward or reverse strand.
#' 
#' @param position The 0-based offset from the start of the forward strand for that reference
#' @param referenceName The name of the reference in whatever reference set is being used
#' @param reverseStrand Whether this position is on the reverse strand, as opposed to the forward
#' 
#' @return Position object
#' 
#' @family Position functions
#' @export
Position <- function(position = NULL, referenceName = NULL, reverseStrand = NULL) {
    structure(list(position = position, referenceName = referenceName, reverseStrand = reverseStrand), 
        class = "gar_Position")
}

#' SearchReferenceSetsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param nextPageToken The continuation token, which is used to page through large result sets
#' @param referenceSets The matching references sets
#' 
#' @return SearchReferenceSetsResponse object
#' 
#' @family SearchReferenceSetsResponse functions
#' @export
SearchReferenceSetsResponse <- function(nextPageToken = NULL, referenceSets = NULL) {
    structure(list(nextPageToken = nextPageToken, referenceSets = referenceSets), 
        class = "gar_SearchReferenceSetsResponse")
}

#' SearchCallSetsRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The call set search request.
#' 
#' @param name Only return call sets for which a substring of the name matches this
#' @param pageToken The continuation token, which is used to page through large result sets
#' @param pageSize The maximum number of results to return in a single page
#' @param variantSetIds Restrict the query to call sets within the given variant sets
#' 
#' @return SearchCallSetsRequest object
#' 
#' @family SearchCallSetsRequest functions
#' @export
SearchCallSetsRequest <- function(name = NULL, pageToken = NULL, pageSize = NULL, 
    variantSetIds = NULL) {
    structure(list(name = name, pageToken = pageToken, pageSize = pageSize, variantSetIds = variantSetIds), 
        class = "gar_SearchCallSetsRequest")
}

#' ImportReadGroupSetsRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The read group set import request.
#' 
#' @param partitionStrategy The partition strategy describes how read groups are partitioned into read
#' @param datasetId Required
#' @param sourceUris A list of URIs pointing at [BAM
#' @param referenceSetId The reference set to which the imported read group sets are aligned to, if
#' 
#' @return ImportReadGroupSetsRequest object
#' 
#' @family ImportReadGroupSetsRequest functions
#' @export
ImportReadGroupSetsRequest <- function(partitionStrategy = NULL, datasetId = NULL, 
    sourceUris = NULL, referenceSetId = NULL) {
    structure(list(partitionStrategy = partitionStrategy, datasetId = datasetId, 
        sourceUris = sourceUris, referenceSetId = referenceSetId), class = "gar_ImportReadGroupSetsRequest")
}

#' Policy Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' Defines an Identity and Access Management (IAM) policy. It is used tospecify access control policies for Cloud Platform resources.A `Policy` consists of a list of `bindings`. A `Binding` binds a list of`members` to a `role`, where the members can be user accounts, Google groups,Google domains, and service accounts. A `role` is a named list of permissionsdefined by IAM.**Example**    {      'bindings': [        {          'role': 'roles/owner',          'members': [            'user:mike@example.com',            'group:admins@example.com',            'domain:google.com',            'serviceAccount:my-other-app@appspot.gserviceaccount.com',          ]        },        {          'role': 'roles/viewer',          'members': ['user:sean@example.com']        }      ]    }For a description of IAM and its features, see the[IAM developer's guide](https://cloud.google.com/iam).
#' 
#' @param etag `etag` is used for optimistic concurrency control as a way to help
#' @param version Version of the `Policy`
#' @param bindings Associates a list of `members` to a `role`
#' 
#' @return Policy object
#' 
#' @family Policy functions
#' @export
Policy <- function(etag = NULL, version = NULL, bindings = NULL) {
    structure(list(etag = etag, version = version, bindings = bindings), class = "gar_Policy")
}

#' Annotation Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' An annotation describes a region of reference genome. The value of anannotation may be one of several canonical types, supplemented by arbitraryinfo tags. An annotation is not inherently associated with a specificsample or individual (though a client could choose to use annotations inthis way). Example canonical annotation types are `GENE` and`VARIANT`.
#' 
#' @param Annotation.info The \link{Annotation.info} object or list of objects
#' @param transcript A transcript value represents the assertion that a particular region of
#' @param start The start position of the range on the reference, 0-based inclusive
#' @param annotationSetId The annotation set to which this annotation belongs
#' @param name The display name of this annotation
#' @param variant A variant annotation, which describes the effect of a variant on the
#' @param referenceId The ID of the Google Genomics reference associated with this range
#' @param id The server-generated annotation ID, unique across all annotations
#' @param reverseStrand Whether this range refers to the reverse strand, as opposed to the forward
#' @param referenceName The display name corresponding to the reference specified by
#' @param info A map of additional read alignment information
#' @param type The data type for this annotation
#' @param end The end position of the range on the reference, 0-based exclusive
#' 
#' @return Annotation object
#' 
#' @family Annotation functions
#' @export
Annotation <- function(Annotation.info = NULL, transcript = NULL, start = NULL, annotationSetId = NULL, 
    name = NULL, variant = NULL, referenceId = NULL, id = NULL, reverseStrand = NULL, 
    referenceName = NULL, info = NULL, type = NULL, end = NULL) {
    structure(list(Annotation.info = Annotation.info, transcript = transcript, start = start, 
        annotationSetId = annotationSetId, name = name, variant = variant, referenceId = referenceId, 
        id = id, reverseStrand = reverseStrand, referenceName = referenceName, info = info, 
        type = type, end = end), class = "gar_Annotation")
}

#' Annotation.info Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A map of additional read alignment information. This must be of the formmap<string, string[]> (string key mapping to a list of string values).
#' 
#' 
#' 
#' @return Annotation.info object
#' 
#' @family Annotation functions
#' @export
Annotation.info <- function() {
    list()
}

#' CancelOperationRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The request message for Operations.CancelOperation.
#' 
#' 
#' 
#' @return CancelOperationRequest object
#' 
#' @family CancelOperationRequest functions
#' @export
CancelOperationRequest <- function() {
    list()
}

#' SearchReadsRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The read search request.
#' 
#' @param start The start position of the range on the reference, 0-based inclusive
#' @param referenceName The reference sequence name, for example `chr1`, `1`, or `chrX`
#' @param readGroupSetIds The IDs of the read groups sets within which to search for reads
#' @param readGroupIds The IDs of the read groups within which to search for reads
#' @param end The end position of the range on the reference, 0-based exclusive
#' @param pageToken The continuation token, which is used to page through large result sets
#' @param pageSize The maximum number of results to return in a single page
#' 
#' @return SearchReadsRequest object
#' 
#' @family SearchReadsRequest functions
#' @export
SearchReadsRequest <- function(start = NULL, referenceName = NULL, readGroupSetIds = NULL, 
    readGroupIds = NULL, end = NULL, pageToken = NULL, pageSize = NULL) {
    structure(list(start = start, referenceName = referenceName, readGroupSetIds = readGroupSetIds, 
        readGroupIds = readGroupIds, end = end, pageToken = pageToken, pageSize = pageSize), 
        class = "gar_SearchReadsRequest")
}

#' RuntimeMetadata Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' Runtime metadata that will be populated in theruntimeMetadatafield of the Operation associated with a RunPipeline execution.
#' 
#' @param computeEngine Execution information specific to Google Compute Engine
#' 
#' @return RuntimeMetadata object
#' 
#' @family RuntimeMetadata functions
#' @export
RuntimeMetadata <- function(computeEngine = NULL) {
    structure(list(computeEngine = computeEngine), class = "gar_RuntimeMetadata")
}

#' Operation Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' This resource represents a long-running operation that is the result of anetwork API call.
#' 
#' @param Operation.response The \link{Operation.response} object or list of objects
#' @param Operation.metadata The \link{Operation.metadata} object or list of objects
#' @param response If importing ReadGroupSets, an ImportReadGroupSetsResponse is returned
#' @param name The server-assigned name, which is only unique within the same service that originally returns it
#' @param error The error result of the operation in case of failure or cancellation
#' @param metadata An OperationMetadata object
#' @param done If the value is `false`, it means the operation is still in progress
#' 
#' @return Operation object
#' 
#' @family Operation functions
#' @export
Operation <- function(Operation.response = NULL, Operation.metadata = NULL, response = NULL, 
    name = NULL, error = NULL, metadata = NULL, done = NULL) {
    structure(list(Operation.response = Operation.response, Operation.metadata = Operation.metadata, 
        response = response, name = name, error = error, metadata = metadata, done = done), 
        class = "gar_Operation")
}

#' Operation.response Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' If importing ReadGroupSets, an ImportReadGroupSetsResponse is returned. If importing Variants, an ImportVariantsResponse is returned. For pipelines and exports, an empty response is returned.
#' 
#' 
#' 
#' @return Operation.response object
#' 
#' @family Operation functions
#' @export
Operation.response <- function() {
    list()
}

#' Operation.metadata Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' An OperationMetadata object. This will always be returned with the Operation.
#' 
#' 
#' 
#' @return Operation.metadata object
#' 
#' @family Operation functions
#' @export
Operation.metadata <- function() {
    list()
}

#' ImportReadGroupSetsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The read group set import response.
#' 
#' @param readGroupSetIds IDs of the read group sets that were created
#' 
#' @return ImportReadGroupSetsResponse object
#' 
#' @family ImportReadGroupSetsResponse functions
#' @export
ImportReadGroupSetsResponse <- function(readGroupSetIds = NULL) {
    structure(list(readGroupSetIds = readGroupSetIds), class = "gar_ImportReadGroupSetsResponse")
}

#' VariantCall Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A call represents the determination of genotype with respect to a particularvariant. It may include associated information such as quality and phasing.For example, a call might assign a probability of 0.32 to the occurrence ofa SNP named rs1234 in a call set with the name NA12345.
#' 
#' @param VariantCall.info The \link{VariantCall.info} object or list of objects
#' @param phaseset If this field is present, this variant call's genotype ordering implies
#' @param info A map of additional variant call information
#' @param callSetName The name of the call set this variant call belongs to
#' @param genotypeLikelihood The genotype likelihoods for this variant call
#' @param callSetId The ID of the call set this variant call belongs to
#' @param genotype The genotype of this variant call
#' 
#' @return VariantCall object
#' 
#' @family VariantCall functions
#' @export
VariantCall <- function(VariantCall.info = NULL, phaseset = NULL, info = NULL, callSetName = NULL, 
    genotypeLikelihood = NULL, callSetId = NULL, genotype = NULL) {
    structure(list(VariantCall.info = VariantCall.info, phaseset = phaseset, info = info, 
        callSetName = callSetName, genotypeLikelihood = genotypeLikelihood, callSetId = callSetId, 
        genotype = genotype), class = "gar_VariantCall")
}

#' VariantCall.info Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A map of additional variant call information. This must be of the formmap<string, string[]> (string key mapping to a list of string values).
#' 
#' 
#' 
#' @return VariantCall.info object
#' 
#' @family VariantCall functions
#' @export
VariantCall.info <- function() {
    list()
}

#' SearchVariantsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The variant search response.
#' 
#' @param nextPageToken The continuation token, which is used to page through large result sets
#' @param variants The list of matching Variants
#' 
#' @return SearchVariantsResponse object
#' 
#' @family SearchVariantsResponse functions
#' @export
SearchVariantsResponse <- function(nextPageToken = NULL, variants = NULL) {
    structure(list(nextPageToken = nextPageToken, variants = variants), class = "gar_SearchVariantsResponse")
}

#' ListBasesResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param sequence A substring of the bases that make up this reference
#' @param offset The offset position (0-based) of the given `sequence` from the
#' @param nextPageToken The continuation token, which is used to page through large result sets
#' 
#' @return ListBasesResponse object
#' 
#' @family ListBasesResponse functions
#' @export
ListBasesResponse <- function(sequence = NULL, offset = NULL, nextPageToken = NULL) {
    structure(list(sequence = sequence, offset = offset, nextPageToken = nextPageToken), 
        class = "gar_ListBasesResponse")
}

#' Status Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The `Status` type defines a logical error model that is suitable for differentprogramming environments, including REST APIs and RPC APIs. It is used by[gRPC](https://github.com/grpc). The error model is designed to be:- Simple to use and understand for most users- Flexible enough to meet unexpected needs# OverviewThe `Status` message contains three pieces of data: error code, error message,and error details. The error code should be an enum value ofgoogle.rpc.Code, but it may accept additional error codes if needed.  Theerror message should be a developer-facing English message that helpsdevelopers *understand* and *resolve* the error. If a localized user-facingerror message is needed, put the localized message in the error details orlocalize it in the client. The optional error details may contain arbitraryinformation about the error. There is a predefined set of error detail typesin the package `google.rpc` which can be used for common error conditions.# Language mappingThe `Status` message is the logical representation of the error model, but itis not necessarily the actual wire format. When the `Status` message isexposed in different client libraries and different wire protocols, it can bemapped differently. For example, it will likely be mapped to some exceptionsin Java, but more likely mapped to some error codes in C.# Other usesThe error model and the `Status` message can be used in a variety ofenvironments, either with or without APIs, to provide aconsistent developer experience across different environments.Example uses of this error model include:- Partial errors. If a service needs to return partial errors to the client,    it may embed the `Status` in the normal response to indicate the partial    errors.- Workflow errors. A typical workflow has multiple steps. Each step may    have a `Status` message for error reporting purpose.- Batch operations. If a client uses batch request and batch response, the    `Status` message should be used directly inside batch response, one for    each error sub-response.- Asynchronous operations. If an API call embeds asynchronous operation    results in its response, the status of those operations should be    represented directly using the `Status` message.- Logging. If some API errors are stored in logs, the message `Status` could    be used directly after any stripping needed for security/privacy reasons.
#' 
#' @param code The status code, which should be an enum value of google
#' @param message A developer-facing error message, which should be in English
#' @param details A list of messages that carry the error details
#' 
#' @return Status object
#' 
#' @family Status functions
#' @export
Status <- function(code = NULL, message = NULL, details = NULL) {
    structure(list(code = code, message = message, details = details), class = "gar_Status")
}

#' Binding Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' Associates `members` with a `role`.
#' 
#' @param role Role that is assigned to `members`
#' @param members Specifies the identities requesting access for a Cloud Platform resource
#' 
#' @return Binding object
#' 
#' @family Binding functions
#' @export
Binding <- function(role = NULL, members = NULL) {
    structure(list(role = role, members = members), class = "gar_Binding")
}

#' UndeleteDatasetRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' 
#' 
#' @return UndeleteDatasetRequest object
#' 
#' @family UndeleteDatasetRequest functions
#' @export
UndeleteDatasetRequest <- function() {
    list()
}

#' Range Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A 0-based half-open genomic coordinate range for search requests.
#' 
#' @param start The start position of the range on the reference, 0-based inclusive
#' @param end The end position of the range on the reference, 0-based exclusive
#' @param referenceName The reference sequence name, for example `chr1`,
#' 
#' @return Range object
#' 
#' @family Range functions
#' @export
Range <- function(start = NULL, end = NULL, referenceName = NULL) {
    structure(list(start = start, end = end, referenceName = referenceName), class = "gar_Range")
}

#' VariantSet Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A variant set is a collection of call sets and variants. It contains summarystatistics of those contents. A variant set belongs to a dataset.For more genomics resource definitions, see [Fundamentals of GoogleGenomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics)
#' 
#' @param name User-specified, mutable name
#' @param referenceSetId The reference set to which the variant set is mapped
#' @param metadata The metadata associated with this variant set
#' @param referenceBounds A list of all references used by the variants in a variant set
#' @param id The server-generated variant set ID, unique across all variant sets
#' @param description A textual description of this variant set
#' @param datasetId The dataset to which this variant set belongs
#' 
#' @return VariantSet object
#' 
#' @family VariantSet functions
#' @export
VariantSet <- function(name = NULL, referenceSetId = NULL, metadata = NULL, referenceBounds = NULL, 
    id = NULL, description = NULL, datasetId = NULL) {
    structure(list(name = name, referenceSetId = referenceSetId, metadata = metadata, 
        referenceBounds = referenceBounds, id = id, description = description, datasetId = datasetId), 
        class = "gar_VariantSet")
}

#' ReferenceBound Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' ReferenceBound records an upper bound for the starting coordinate ofvariants in a particular reference.
#' 
#' @param upperBound An upper bound (inclusive) on the starting coordinate of any
#' @param referenceName The name of the reference associated with this reference bound
#' 
#' @return ReferenceBound object
#' 
#' @family ReferenceBound functions
#' @export
ReferenceBound <- function(upperBound = NULL, referenceName = NULL) {
    structure(list(upperBound = upperBound, referenceName = referenceName), class = "gar_ReferenceBound")
}

#' BatchCreateAnnotationsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param entries The resulting per-annotation entries, ordered consistently with the
#' 
#' @return BatchCreateAnnotationsResponse object
#' 
#' @family BatchCreateAnnotationsResponse functions
#' @export
BatchCreateAnnotationsResponse <- function(entries = NULL) {
    structure(list(entries = entries), class = "gar_BatchCreateAnnotationsResponse")
}

#' Variant Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A variant represents a change in DNA sequence relative to a referencesequence. For example, a variant could represent a SNP or an insertion.Variants belong to a variant set.For more genomics resource definitions, see [Fundamentals of GoogleGenomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics)Each of the calls on a variant represent a determination of genotype withrespect to that variant. For example, a call might assign probability of 0.32to the occurrence of a SNP named rs1234 in a sample named NA12345. A callbelongs to a call set, which contains related calls typically from onesample.
#' 
#' @param Variant.info The \link{Variant.info} object or list of objects
#' @param variantSetId The ID of the variant set this variant belongs to
#' @param referenceName The reference on which this variant occurs
#' @param info A map of additional variant information
#' @param referenceBases The reference bases for this variant
#' @param alternateBases The bases that appear instead of the reference bases
#' @param names Names for the variant, for example a RefSNP ID
#' @param end The end position (0-based) of this variant
#' @param filter A list of filters (normally quality filters) this variant has failed
#' @param calls The variant calls for this particular variant
#' @param created The date this variant was created, in milliseconds from the epoch
#' @param start The position at which this variant occurs (0-based)
#' @param quality A measure of how likely this variant is to be real
#' @param id The server-generated variant ID, unique across all variants
#' 
#' @return Variant object
#' 
#' @family Variant functions
#' @export
Variant <- function(Variant.info = NULL, variantSetId = NULL, referenceName = NULL, 
    info = NULL, referenceBases = NULL, alternateBases = NULL, names = NULL, end = NULL, 
    filter = NULL, calls = NULL, created = NULL, start = NULL, quality = NULL, id = NULL) {
    structure(list(Variant.info = Variant.info, variantSetId = variantSetId, referenceName = referenceName, 
        info = info, referenceBases = referenceBases, alternateBases = alternateBases, 
        names = names, end = end, filter = filter, calls = calls, created = created, 
        start = start, quality = quality, id = id), class = "gar_Variant")
}

#' Variant.info Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A map of additional variant information. This must be of the formmap<string, string[]> (string key mapping to a list of string values).
#' 
#' 
#' 
#' @return Variant.info object
#' 
#' @family Variant functions
#' @export
Variant.info <- function() {
    list()
}

#' ListOperationsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The response message for Operations.ListOperations.
#' 
#' @param operations A list of operations that matches the specified filter in the request
#' @param nextPageToken The standard List next-page token
#' 
#' @return ListOperationsResponse object
#' 
#' @family ListOperationsResponse functions
#' @export
ListOperationsResponse <- function(operations = NULL, nextPageToken = NULL) {
    structure(list(operations = operations, nextPageToken = nextPageToken), class = "gar_ListOperationsResponse")
}

#' SearchCallSetsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The call set search response.
#' 
#' @param nextPageToken The continuation token, which is used to page through large result sets
#' @param callSets The list of matching call sets
#' 
#' @return SearchCallSetsResponse object
#' 
#' @family SearchCallSetsResponse functions
#' @export
SearchCallSetsResponse <- function(nextPageToken = NULL, callSets = NULL) {
    structure(list(nextPageToken = nextPageToken, callSets = callSets), class = "gar_SearchCallSetsResponse")
}

#' SearchVariantsRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The variant search request.
#' 
#' @param referenceName Required
#' @param variantSetIds At most one variant set ID must be provided
#' @param end The end of the window, 0-based exclusive
#' @param pageToken The continuation token, which is used to page through large result sets
#' @param maxCalls The maximum number of calls to return in a single page
#' @param pageSize The maximum number of variants to return in a single page
#' @param callSetIds Only return variant calls which belong to call sets with these ids
#' @param variantName Only return variants which have exactly this name
#' @param start The beginning of the window (0-based, inclusive) for which
#' 
#' @return SearchVariantsRequest object
#' 
#' @family SearchVariantsRequest functions
#' @export
SearchVariantsRequest <- function(referenceName = NULL, variantSetIds = NULL, end = NULL, 
    pageToken = NULL, maxCalls = NULL, pageSize = NULL, callSetIds = NULL, variantName = NULL, 
    start = NULL) {
    structure(list(referenceName = referenceName, variantSetIds = variantSetIds, 
        end = end, pageToken = pageToken, maxCalls = maxCalls, pageSize = pageSize, 
        callSetIds = callSetIds, variantName = variantName, start = start), class = "gar_SearchVariantsRequest")
}

#' OperationMetadata Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' Metadata describing an Operation.
#' 
#' @param OperationMetadata.request The \link{OperationMetadata.request} object or list of objects
#' @param OperationMetadata.runtimeMetadata The \link{OperationMetadata.runtimeMetadata} object or list of objects
#' @param OperationMetadata.labels The \link{OperationMetadata.labels} object or list of objects
#' @param projectId The Google Cloud Project in which the job is scoped
#' @param clientId This field is deprecated
#' @param events Optional event messages that were generated during the job's execution
#' @param endTime The time at which the job stopped running
#' @param startTime The time at which the job began to run
#' @param request The original request that started the operation
#' @param runtimeMetadata Runtime metadata on this Operation
#' @param createTime The time at which the job was submitted to the Genomics service
#' @param labels Optionally provided by the caller when submitting the request that creates
#' 
#' @return OperationMetadata object
#' 
#' @family OperationMetadata functions
#' @export
OperationMetadata <- function(OperationMetadata.request = NULL, OperationMetadata.runtimeMetadata = NULL, 
    OperationMetadata.labels = NULL, projectId = NULL, clientId = NULL, events = NULL, 
    endTime = NULL, startTime = NULL, request = NULL, runtimeMetadata = NULL, createTime = NULL, 
    labels = NULL) {
    structure(list(OperationMetadata.request = OperationMetadata.request, OperationMetadata.runtimeMetadata = OperationMetadata.runtimeMetadata, 
        OperationMetadata.labels = OperationMetadata.labels, projectId = projectId, 
        clientId = clientId, events = events, endTime = endTime, startTime = startTime, 
        request = request, runtimeMetadata = runtimeMetadata, createTime = createTime, 
        labels = labels), class = "gar_OperationMetadata")
}

#' OperationMetadata.request Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The original request that started the operation. Note that this will be incurrent version of the API. If the operation was started with v1beta2 APIand a GetOperation is performed on v1 API, a v1 request will be returned.
#' 
#' 
#' 
#' @return OperationMetadata.request object
#' 
#' @family OperationMetadata functions
#' @export
OperationMetadata.request <- function() {
    list()
}

#' OperationMetadata.runtimeMetadata Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' Runtime metadata on this Operation.
#' 
#' 
#' 
#' @return OperationMetadata.runtimeMetadata object
#' 
#' @family OperationMetadata functions
#' @export
OperationMetadata.runtimeMetadata <- function() {
    list()
}

#' OperationMetadata.labels Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' Optionally provided by the caller when submitting the request that createsthe operation.
#' 
#' 
#' 
#' @return OperationMetadata.labels object
#' 
#' @family OperationMetadata functions
#' @export
OperationMetadata.labels <- function() {
    list()
}

#' SearchReadGroupSetsRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The read group set search request.
#' 
#' @param datasetIds Restricts this query to read group sets within the given datasets
#' @param name Only return read group sets for which a substring of the name matches this
#' @param pageToken The continuation token, which is used to page through large result sets
#' @param pageSize The maximum number of results to return in a single page
#' 
#' @return SearchReadGroupSetsRequest object
#' 
#' @family SearchReadGroupSetsRequest functions
#' @export
SearchReadGroupSetsRequest <- function(datasetIds = NULL, name = NULL, pageToken = NULL, 
    pageSize = NULL) {
    structure(list(datasetIds = datasetIds, name = name, pageToken = pageToken, pageSize = pageSize), 
        class = "gar_SearchReadGroupSetsRequest")
}

#' SearchAnnotationsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param nextPageToken The continuation token, which is used to page through large result sets
#' @param annotations The matching annotations
#' 
#' @return SearchAnnotationsResponse object
#' 
#' @family SearchAnnotationsResponse functions
#' @export
SearchAnnotationsResponse <- function(nextPageToken = NULL, annotations = NULL) {
    structure(list(nextPageToken = nextPageToken, annotations = annotations), class = "gar_SearchAnnotationsResponse")
}

#' ClinicalCondition Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param conceptId The MedGen concept id associated with this gene
#' @param names A set of names for the condition
#' @param omimId The OMIM id for this condition
#' @param externalIds The set of external IDs for this condition
#' 
#' @return ClinicalCondition object
#' 
#' @family ClinicalCondition functions
#' @export
ClinicalCondition <- function(conceptId = NULL, names = NULL, omimId = NULL, externalIds = NULL) {
    structure(list(conceptId = conceptId, names = names, omimId = omimId, externalIds = externalIds), 
        class = "gar_ClinicalCondition")
}

#' SearchReadsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The read search response.
#' 
#' @param alignments The list of matching alignments sorted by mapped genomic coordinate,
#' @param nextPageToken The continuation token, which is used to page through large result sets
#' 
#' @return SearchReadsResponse object
#' 
#' @family SearchReadsResponse functions
#' @export
SearchReadsResponse <- function(alignments = NULL, nextPageToken = NULL) {
    structure(list(alignments = alignments, nextPageToken = nextPageToken), class = "gar_SearchReadsResponse")
}

#' Program Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param commandLine The command line used to run this program
#' @param prevProgramId The ID of the program run before this one
#' @param id The user specified locally unique ID of the program
#' @param version The version of the program run
#' @param name The display name of the program
#' 
#' @return Program object
#' 
#' @family Program functions
#' @export
Program <- function(commandLine = NULL, prevProgramId = NULL, id = NULL, version = NULL, 
    name = NULL) {
    structure(list(commandLine = commandLine, prevProgramId = prevProgramId, id = id, 
        version = version, name = name), class = "gar_Program")
}

#' ComputeEngine Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' Describes a Compute Engine resource that is being managed by a runningpipeline.
#' 
#' @param zone The availability zone in which the instance resides
#' @param machineType The machine type of the instance
#' @param diskNames The names of the disks that were created for this pipeline
#' @param instanceName The instance on which the operation is running
#' 
#' @return ComputeEngine object
#' 
#' @family ComputeEngine functions
#' @export
ComputeEngine <- function(zone = NULL, machineType = NULL, diskNames = NULL, instanceName = NULL) {
    structure(list(zone = zone, machineType = machineType, diskNames = diskNames, 
        instanceName = instanceName), class = "gar_ComputeEngine")
}

#' CoverageBucket Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A bucket over which read coverage has been precomputed. A bucket correspondsto a specific range of the reference sequence.
#' 
#' @param meanCoverage The average number of reads which are aligned to each individual
#' @param range The genomic coordinate range spanned by this bucket
#' 
#' @return CoverageBucket object
#' 
#' @family CoverageBucket functions
#' @export
CoverageBucket <- function(meanCoverage = NULL, range = NULL) {
    structure(list(meanCoverage = meanCoverage, range = range), class = "gar_CoverageBucket")
}

#' ExternalId Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param sourceName The name of the source of this data
#' @param id The id used by the source of this data
#' 
#' @return ExternalId object
#' 
#' @family ExternalId functions
#' @export
ExternalId <- function(sourceName = NULL, id = NULL) {
    structure(list(sourceName = sourceName, id = id), class = "gar_ExternalId")
}

#' SearchVariantSetsRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The search variant sets request.
#' 
#' @param pageToken The continuation token, which is used to page through large result sets
#' @param pageSize The maximum number of results to return in a single page
#' @param datasetIds Exactly one dataset ID must be provided here
#' 
#' @return SearchVariantSetsRequest object
#' 
#' @family SearchVariantSetsRequest functions
#' @export
SearchVariantSetsRequest <- function(pageToken = NULL, pageSize = NULL, datasetIds = NULL) {
    structure(list(pageToken = pageToken, pageSize = pageSize, datasetIds = datasetIds), 
        class = "gar_SearchVariantSetsRequest")
}

#' VariantSetMetadata Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' Metadata describes a single piece of variant call metadata.These data include a top level key and either a single value string (value)or a list of key-value pairs (info.)Value and info are mutually exclusive.
#' 
#' @param VariantSetMetadata.info The \link{VariantSetMetadata.info} object or list of objects
#' @param info Remaining structured metadata key-value pairs
#' @param type The type of data
#' @param number The number of values that can be included in a field described by this
#' @param id User-provided ID field, not enforced by this API
#' @param value The value field for simple metadata
#' @param key The top-level key
#' @param description A textual description of this metadata
#' 
#' @return VariantSetMetadata object
#' 
#' @family VariantSetMetadata functions
#' @export
VariantSetMetadata <- function(VariantSetMetadata.info = NULL, info = NULL, type = NULL, 
    number = NULL, id = NULL, value = NULL, key = NULL, description = NULL) {
    structure(list(VariantSetMetadata.info = VariantSetMetadata.info, info = info, 
        type = type, number = number, id = id, value = value, key = key, description = description), 
        class = "gar_VariantSetMetadata")
}

#' VariantSetMetadata.info Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' Remaining structured metadata key-value pairs. This must be of the formmap<string, string[]> (string key mapping to a list of string values).
#' 
#' 
#' 
#' @return VariantSetMetadata.info object
#' 
#' @family VariantSetMetadata functions
#' @export
VariantSetMetadata.info <- function() {
    list()
}

#' Reference Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A reference is a canonical assembled DNA sequence, intended to act as areference coordinate space for other genomic annotations. A single referencemight represent the human chromosome 1 or mitochandrial DNA, for instance. Areference belongs to one or more reference sets.For more genomics resource definitions, see [Fundamentals of GoogleGenomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics)
#' 
#' @param ncbiTaxonId ID from http://www
#' @param sourceUri The URI from which the sequence was obtained
#' @param name The name of this reference, for example `22`
#' @param md5checksum MD5 of the upper-case sequence excluding all whitespace characters (this
#' @param id The server-generated reference ID, unique across all references
#' @param length The length of this reference's sequence
#' @param sourceAccessions All known corresponding accession IDs in INSDC (GenBank/ENA/DDBJ) ideally
#' 
#' @return Reference object
#' 
#' @family Reference functions
#' @export
Reference <- function(ncbiTaxonId = NULL, sourceUri = NULL, name = NULL, md5checksum = NULL, 
    id = NULL, length = NULL, sourceAccessions = NULL) {
    structure(list(ncbiTaxonId = ncbiTaxonId, sourceUri = sourceUri, name = name, 
        md5checksum = md5checksum, id = id, length = length, sourceAccessions = sourceAccessions), 
        class = "gar_Reference")
}

#' SearchReferenceSetsRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param assemblyId If present, return reference sets for which a substring of their
#' @param md5checksums If present, return reference sets for which the
#' @param accessions If present, return reference sets for which a prefix of any of
#' @param pageToken The continuation token, which is used to page through large result sets
#' @param pageSize The maximum number of results to return in a single page
#' 
#' @return SearchReferenceSetsRequest object
#' 
#' @family SearchReferenceSetsRequest functions
#' @export
SearchReferenceSetsRequest <- function(assemblyId = NULL, md5checksums = NULL, accessions = NULL, 
    pageToken = NULL, pageSize = NULL) {
    structure(list(assemblyId = assemblyId, md5checksums = md5checksums, accessions = accessions, 
        pageToken = pageToken, pageSize = pageSize), class = "gar_SearchReferenceSetsRequest")
}

#' SetIamPolicyRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' Request message for `SetIamPolicy` method.
#' 
#' @param policy REQUIRED: The complete policy to be applied to the `resource`
#' 
#' @return SetIamPolicyRequest object
#' 
#' @family SetIamPolicyRequest functions
#' @export
SetIamPolicyRequest <- function(policy = NULL) {
    structure(list(policy = policy), class = "gar_SetIamPolicyRequest")
}

#' MergeVariantsRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param MergeVariantsRequest.infoMergeConfig The \link{MergeVariantsRequest.infoMergeConfig} object or list of objects
#' @param variantSetId The destination variant set
#' @param variants The variants to be merged with existing variants
#' @param infoMergeConfig A mapping between info field keys and the InfoMergeOperations to
#' 
#' @return MergeVariantsRequest object
#' 
#' @family MergeVariantsRequest functions
#' @export
MergeVariantsRequest <- function(MergeVariantsRequest.infoMergeConfig = NULL, variantSetId = NULL, 
    variants = NULL, infoMergeConfig = NULL) {
    structure(list(MergeVariantsRequest.infoMergeConfig = MergeVariantsRequest.infoMergeConfig, 
        variantSetId = variantSetId, variants = variants, infoMergeConfig = infoMergeConfig), 
        class = "gar_MergeVariantsRequest")
}

#' MergeVariantsRequest.infoMergeConfig Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A mapping between info field keys and the InfoMergeOperations tobe performed on them.
#' 
#' 
#' 
#' @return MergeVariantsRequest.infoMergeConfig object
#' 
#' @family MergeVariantsRequest functions
#' @export
MergeVariantsRequest.infoMergeConfig <- function() {
    list()
}

#' Read Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A read alignment describes a linear alignment of a string of DNA to areference sequence, in addition to metadataabout the fragment (the molecule of DNA sequenced) and the read (the baseswhich were read by the sequencer). A read is equivalent to a line in a SAMfile. A read belongs to exactly one read group and exactly oneread group set.For more genomics resource definitions, see [Fundamentals of GoogleGenomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics)### Reverse-stranded readsMapped reads (reads having a non-null `alignment`) can be aligned to eitherthe forward or the reverse strand of their associated reference. Strandednessof a mapped read is encoded by `alignment.position.reverseStrand`.If we consider the reference to be a forward-stranded coordinate space of`[0, reference.length)` with `0` as the left-most position and`reference.length` as the right-most position, reads are always aligned leftto right. That is, `alignment.position.position` always refers to theleft-most reference coordinate and `alignment.cigar` describes the alignmentof this read to the reference from left to right. All per-base fields such as`alignedSequence` and `alignedQuality` share this same left-to-rightorientation; this is true of reads which are aligned to either strand. Forreverse-stranded reads, this means that `alignedSequence` is the reversecomplement of the bases that were originally reported by the sequencingmachine.### Generating a reference-aligned sequence stringWhen interacting with mapped reads, it's often useful to produce a stringrepresenting the local alignment of the read to reference. The followingpseudocode demonstrates one way of doing this:    out = ''    offset = 0    for c in read.alignment.cigar {      switch c.operation {      case 'ALIGNMENT_MATCH', 'SEQUENCE_MATCH', 'SEQUENCE_MISMATCH':        out += read.alignedSequence[offset:offset+c.operationLength]        offset += c.operationLength        break      case 'CLIP_SOFT', 'INSERT':        offset += c.operationLength        break      case 'PAD':        out += repeat('*', c.operationLength)        break      case 'DELETE':        out += repeat('-', c.operationLength)        break      case 'SKIP':        out += repeat(' ', c.operationLength)        break      case 'CLIP_HARD':        break      }    }    return out### Converting to SAM's CIGAR stringThe following pseudocode generates a SAM CIGAR string from the`cigar` field. Note that this is a lossy conversion(`cigar.referenceSequence` is lost).    cigarMap = {      'ALIGNMENT_MATCH': 'M',      'INSERT': 'I',      'DELETE': 'D',      'SKIP': 'N',      'CLIP_SOFT': 'S',      'CLIP_HARD': 'H',      'PAD': 'P',      'SEQUENCE_MATCH': '=',      'SEQUENCE_MISMATCH': 'X',    }    cigarStr = ''    for c in read.alignment.cigar {      cigarStr += c.operationLength + cigarMap[c.operation]    }    return cigarStr
#' 
#' @param Read.info The \link{Read.info} object or list of objects
#' @param fragmentName The fragment name
#' @param readGroupSetId The ID of the read group set this read belongs to
#' @param duplicateFragment The fragment is a PCR or optical duplicate (SAM flag 0x400)
#' @param readNumber The read number in sequencing
#' @param readGroupId The ID of the read group this read belongs to
#' @param alignedSequence The bases of the read sequence contained in this alignment record,
#' @param info A map of additional read alignment information
#' @param nextMatePosition The mapping of the primary alignment of the
#' @param supplementaryAlignment Whether this alignment is supplementary
#' @param properPlacement The orientation and the distance between reads from the fragment are
#' @param fragmentLength The observed length of the fragment, equivalent to TLEN in SAM
#' @param failedVendorQualityChecks Whether this read did not pass filters, such as platform or vendor quality
#' @param alignedQuality The quality of the read sequence contained in this alignment record
#' @param alignment The linear alignment for this alignment record
#' @param numberReads The number of reads in the fragment (extension to SAM flag 0x1)
#' @param id The server-generated read ID, unique across all reads
#' @param secondaryAlignment Whether this alignment is secondary
#' 
#' @return Read object
#' 
#' @family Read functions
#' @export
Read <- function(Read.info = NULL, fragmentName = NULL, readGroupSetId = NULL, duplicateFragment = NULL, 
    readNumber = NULL, readGroupId = NULL, alignedSequence = NULL, info = NULL, nextMatePosition = NULL, 
    supplementaryAlignment = NULL, properPlacement = NULL, fragmentLength = NULL, 
    failedVendorQualityChecks = NULL, alignedQuality = NULL, alignment = NULL, numberReads = NULL, 
    id = NULL, secondaryAlignment = NULL) {
    structure(list(Read.info = Read.info, fragmentName = fragmentName, readGroupSetId = readGroupSetId, 
        duplicateFragment = duplicateFragment, readNumber = readNumber, readGroupId = readGroupId, 
        alignedSequence = alignedSequence, info = info, nextMatePosition = nextMatePosition, 
        supplementaryAlignment = supplementaryAlignment, properPlacement = properPlacement, 
        fragmentLength = fragmentLength, failedVendorQualityChecks = failedVendorQualityChecks, 
        alignedQuality = alignedQuality, alignment = alignment, numberReads = numberReads, 
        id = id, secondaryAlignment = secondaryAlignment), class = "gar_Read")
}

#' Read.info Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A map of additional read alignment information. This must be of the formmap<string, string[]> (string key mapping to a list of string values).
#' 
#' 
#' 
#' @return Read.info object
#' 
#' @family Read functions
#' @export
Read.info <- function() {
    list()
}

#' BatchCreateAnnotationsRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param requestId A unique request ID which enables the server to detect duplicated requests
#' @param annotations The annotations to be created
#' 
#' @return BatchCreateAnnotationsRequest object
#' 
#' @family BatchCreateAnnotationsRequest functions
#' @export
BatchCreateAnnotationsRequest <- function(requestId = NULL, annotations = NULL) {
    structure(list(requestId = requestId, annotations = annotations), class = "gar_BatchCreateAnnotationsRequest")
}

#' CigarUnit Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A single CIGAR operation.
#' 
#' @param operation No description
#' @param referenceSequence `referenceSequence` is only used at mismatches
#' @param operationLength The number of genomic bases that the operation runs for
#' 
#' @return CigarUnit object
#' 
#' @family CigarUnit functions
#' @export
CigarUnit <- function(operation = NULL, referenceSequence = NULL, operationLength = NULL) {
    structure(list(operation = operation, referenceSequence = referenceSequence, 
        operationLength = operationLength), class = "gar_CigarUnit")
}

#' ReferenceSet Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A reference set is a set of references which typically comprise a referenceassembly for a species, such as `GRCh38` which is representativeof the human genome. A reference set defines a common coordinate space forcomparing reference-aligned experimental data. A reference set contains 1 ormore references.For more genomics resource definitions, see [Fundamentals of GoogleGenomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics)
#' 
#' @param description Free text description of this reference set
#' @param sourceAccessions All known corresponding accession IDs in INSDC (GenBank/ENA/DDBJ) ideally
#' @param ncbiTaxonId ID from http://www
#' @param sourceUri The URI from which the references were obtained
#' @param referenceIds The IDs of the reference objects that are part of this set
#' @param md5checksum Order-independent MD5 checksum which identifies this reference set
#' @param assemblyId Public id of this reference set, such as `GRCh37`
#' @param id The server-generated reference set ID, unique across all reference sets
#' 
#' @return ReferenceSet object
#' 
#' @family ReferenceSet functions
#' @export
ReferenceSet <- function(description = NULL, sourceAccessions = NULL, ncbiTaxonId = NULL, 
    sourceUri = NULL, referenceIds = NULL, md5checksum = NULL, assemblyId = NULL, 
    id = NULL) {
    structure(list(description = description, sourceAccessions = sourceAccessions, 
        ncbiTaxonId = ncbiTaxonId, sourceUri = sourceUri, referenceIds = referenceIds, 
        md5checksum = md5checksum, assemblyId = assemblyId, id = id), class = "gar_ReferenceSet")
}

#' Transcript Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A transcript represents the assertion that a particular region of thereference genome may be transcribed as RNA.
#' 
#' @param geneId The annotation ID of the gene from which this transcript is transcribed
#' @param exons The <a href='http://en
#' @param codingSequence The range of the coding sequence for this transcript, if any
#' 
#' @return Transcript object
#' 
#' @family Transcript functions
#' @export
Transcript <- function(geneId = NULL, exons = NULL, codingSequence = NULL) {
    structure(list(geneId = geneId, exons = exons, codingSequence = codingSequence), 
        class = "gar_Transcript")
}

#' AnnotationSet Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' An annotation set is a logical grouping of annotations that share consistenttype information and provenance. Examples of annotation sets include 'allgenes from refseq', and 'all variant annotations from ClinVar'.
#' 
#' @param AnnotationSet.info The \link{AnnotationSet.info} object or list of objects
#' @param datasetId The dataset to which this annotation set belongs
#' @param sourceUri The source URI describing the file from which this annotation set was
#' @param name The display name for this annotation set
#' @param referenceSetId The ID of the reference set that defines the coordinate space for this
#' @param info A map of additional read alignment information
#' @param type The type of annotations contained within this set
#' @param id The server-generated annotation set ID, unique across all annotation sets
#' 
#' @return AnnotationSet object
#' 
#' @family AnnotationSet functions
#' @export
AnnotationSet <- function(AnnotationSet.info = NULL, datasetId = NULL, sourceUri = NULL, 
    name = NULL, referenceSetId = NULL, info = NULL, type = NULL, id = NULL) {
    structure(list(AnnotationSet.info = AnnotationSet.info, datasetId = datasetId, 
        sourceUri = sourceUri, name = name, referenceSetId = referenceSetId, info = info, 
        type = type, id = id), class = "gar_AnnotationSet")
}

#' AnnotationSet.info Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A map of additional read alignment information. This must be of the formmap<string, string[]> (string key mapping to a list of string values).
#' 
#' 
#' 
#' @return AnnotationSet.info object
#' 
#' @family AnnotationSet functions
#' @export
AnnotationSet.info <- function() {
    list()
}

#' Experiment Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param platformUnit The platform unit used as part of this experiment, for example
#' @param libraryId A client-supplied library identifier; a library is a collection of DNA
#' @param instrumentModel The instrument model used as part of this experiment
#' @param sequencingCenter The sequencing center used as part of this experiment
#' 
#' @return Experiment object
#' 
#' @family Experiment functions
#' @export
Experiment <- function(platformUnit = NULL, libraryId = NULL, instrumentModel = NULL, 
    sequencingCenter = NULL) {
    structure(list(platformUnit = platformUnit, libraryId = libraryId, instrumentModel = instrumentModel, 
        sequencingCenter = sequencingCenter), class = "gar_Experiment")
}

#' ListDatasetsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The dataset list response.
#' 
#' @param datasets The list of matching Datasets
#' @param nextPageToken The continuation token, which is used to page through large result sets
#' 
#' @return ListDatasetsResponse object
#' 
#' @family ListDatasetsResponse functions
#' @export
ListDatasetsResponse <- function(datasets = NULL, nextPageToken = NULL) {
    structure(list(datasets = datasets, nextPageToken = nextPageToken), class = "gar_ListDatasetsResponse")
}

#' TestIamPermissionsRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' Request message for `TestIamPermissions` method.
#' 
#' @param permissions REQUIRED: The set of permissions to check for the 'resource'
#' 
#' @return TestIamPermissionsRequest object
#' 
#' @family TestIamPermissionsRequest functions
#' @export
TestIamPermissionsRequest <- function(permissions = NULL) {
    structure(list(permissions = permissions), class = "gar_TestIamPermissionsRequest")
}

#' Exon Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param start The start position of the exon on this annotation's reference sequence,
#' @param end The end position of the exon on this annotation's reference sequence,
#' @param frame The frame of this exon
#' 
#' @return Exon object
#' 
#' @family Exon functions
#' @export
Exon <- function(start = NULL, end = NULL, frame = NULL) {
    structure(list(start = start, end = end, frame = frame), class = "gar_Exon")
}

#' ExportReadGroupSetRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The read group set export request.
#' 
#' @param exportUri Required
#' @param referenceNames The reference names to export
#' @param projectId Required
#' 
#' @return ExportReadGroupSetRequest object
#' 
#' @family ExportReadGroupSetRequest functions
#' @export
ExportReadGroupSetRequest <- function(exportUri = NULL, referenceNames = NULL, projectId = NULL) {
    structure(list(exportUri = exportUri, referenceNames = referenceNames, projectId = projectId), 
        class = "gar_ExportReadGroupSetRequest")
}

#' CallSet Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A call set is a collection of variant calls, typically for one sample. Itbelongs to a variant set.For more genomics resource definitions, see [Fundamentals of GoogleGenomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics)
#' 
#' @param CallSet.info The \link{CallSet.info} object or list of objects
#' @param sampleId The sample ID this call set corresponds to
#' @param name The call set name
#' @param info A map of additional call set information
#' @param variantSetIds The IDs of the variant sets this call set belongs to
#' @param id The server-generated call set ID, unique across all call sets
#' @param created The date this call set was created in milliseconds from the epoch
#' 
#' @return CallSet object
#' 
#' @family CallSet functions
#' @export
CallSet <- function(CallSet.info = NULL, sampleId = NULL, name = NULL, info = NULL, 
    variantSetIds = NULL, id = NULL, created = NULL) {
    structure(list(CallSet.info = CallSet.info, sampleId = sampleId, name = name, 
        info = info, variantSetIds = variantSetIds, id = id, created = created), 
        class = "gar_CallSet")
}

#' CallSet.info Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A map of additional call set information. This must be of the formmap<string, string[]> (string key mapping to a list of string values).
#' 
#' 
#' 
#' @return CallSet.info object
#' 
#' @family CallSet functions
#' @export
CallSet.info <- function() {
    list()
}

#' SearchAnnotationSetsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param nextPageToken The continuation token, which is used to page through large result sets
#' @param annotationSets The matching annotation sets
#' 
#' @return SearchAnnotationSetsResponse object
#' 
#' @family SearchAnnotationSetsResponse functions
#' @export
SearchAnnotationSetsResponse <- function(nextPageToken = NULL, annotationSets = NULL) {
    structure(list(nextPageToken = nextPageToken, annotationSets = annotationSets), 
        class = "gar_SearchAnnotationSetsResponse")
}

#' ImportVariantsRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The variant data import request.
#' 
#' @param ImportVariantsRequest.infoMergeConfig The \link{ImportVariantsRequest.infoMergeConfig} object or list of objects
#' @param variantSetId Required
#' @param sourceUris A list of URIs referencing variant files in Google Cloud Storage
#' @param normalizeReferenceNames Convert reference names to the canonical representation
#' @param format The format of the variant data being imported
#' @param infoMergeConfig A mapping between info field keys and the InfoMergeOperations to
#' 
#' @return ImportVariantsRequest object
#' 
#' @family ImportVariantsRequest functions
#' @export
ImportVariantsRequest <- function(ImportVariantsRequest.infoMergeConfig = NULL, variantSetId = NULL, 
    sourceUris = NULL, normalizeReferenceNames = NULL, format = NULL, infoMergeConfig = NULL) {
    structure(list(ImportVariantsRequest.infoMergeConfig = ImportVariantsRequest.infoMergeConfig, 
        variantSetId = variantSetId, sourceUris = sourceUris, normalizeReferenceNames = normalizeReferenceNames, 
        format = format, infoMergeConfig = infoMergeConfig), class = "gar_ImportVariantsRequest")
}

#' ImportVariantsRequest.infoMergeConfig Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A mapping between info field keys and the InfoMergeOperations tobe performed on them. This is plumbed down to the MergeVariantRequestsgenerated by the resulting import job.
#' 
#' 
#' 
#' @return ImportVariantsRequest.infoMergeConfig object
#' 
#' @family ImportVariantsRequest functions
#' @export
ImportVariantsRequest.infoMergeConfig <- function() {
    list()
}

#' ListCoverageBucketsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param nextPageToken The continuation token, which is used to page through large result sets
#' @param bucketWidth The length of each coverage bucket in base pairs
#' @param coverageBuckets The coverage buckets
#' 
#' @return ListCoverageBucketsResponse object
#' 
#' @family ListCoverageBucketsResponse functions
#' @export
ListCoverageBucketsResponse <- function(nextPageToken = NULL, bucketWidth = NULL, 
    coverageBuckets = NULL) {
    structure(list(nextPageToken = nextPageToken, bucketWidth = bucketWidth, coverageBuckets = coverageBuckets), 
        class = "gar_ListCoverageBucketsResponse")
}


#' VariantAnnotation Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param conditions The set of conditions associated with this variant
#' @param effect Effect of the variant on the coding sequence
#' @param transcriptIds Google annotation IDs of the transcripts affected by this variant
#' @param type Type has been adapted from ClinVar's list of variant types
#' @param alternateBases The alternate allele for this variant
#' @param geneId Google annotation ID of the gene affected by this variant
#' @param clinicalSignificance Describes the clinical significance of a variant
#' 
#' @return VariantAnnotation object
#' 
#' @family VariantAnnotation functions
#' @export


VariantAnnotation <- function(conditions = NULL, effect = NULL, transcriptIds = NULL, 
    type = NULL, alternateBases = NULL, geneId = NULL, clinicalSignificance = NULL) {
    
    
    
    structure(list(conditions = conditions, effect = effect, transcriptIds = transcriptIds, 
        type = type, alternateBases = alternateBases, geneId = geneId, clinicalSignificance = clinicalSignificance), 
        class = "gar_VariantAnnotation")
}

